unittests/test_modules.py

```diff
@@ -402,3 +402,45 @@ def test_mapping_deep_cycle(self):
         }
         self.assertRaisesRegex(EnvironError, 'm0->m1->m3->m4->m2->m1',
                                self.modules_system.load_module, 'm0')
+
+    def test_mapping_from_file_simple(self):
+        with open('module_mappings_file.txt', 'w') as f:
```

Comment:
You should create a temporary file for that. Also use `fp` instead of `f`, since this is the "convention" elsewhere in the code.